aws/resource_aws_appsync_graphql_api.go

```diff
@@ -167,7 +166,7 @@ func resourceAwsAppsyncGraphqlApiCreate(d *schema.ResourceData, meta interface{}
 
```

Comment:
Not related to this pull request -- when running the acceptance testing for `aws_appsync_graphql_api`, was hitting this pretty often with random tests and our default 20 concurrency:

```
--- FAIL: TestAccAWSAppsyncGraphqlApi_Schema (11.25s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
          * aws_appsync_graphql_api.test: 1 error occurred:
          * aws_appsync_graphql_api.test: ConcurrentModificationException: Can't create new GraphQL API, a GraphQL API creation is already in progress.

--- FAIL: TestAccAWSAppsyncGraphqlApi_OpenIDConnectConfig_ClientID (14.78s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
          * aws_appsync_graphql_api.test: 1 error occurred:
          * aws_appsync_graphql_api.test: ConcurrentModificationException: Can't create new GraphQL API, a GraphQL API creation is already in progress.
```

Judging by the lack of error context before `ConcurrentModificationException` here and from the debug logs this was occurring during the `CreateGraphqlApi` call. In a followup commit to ensure a great user experience, I went ahead and added the error context to that call so we can know where this similar looking error is happening (`return fmt.Errorf("error creating AppSync GraphQL API: %d", err)`) and updated the AWS Go SDK AppSync client to automatically enable retries for that call in `aws/config.go`:

```go
	client.appsyncconn.Handlers.Retry.PushBack(func(r *request.Request) {
		if r.Operation.Name == "CreateGraphqlApi" {
			if isAWSErr(r.Error, appsync.ErrCodeConcurrentModificationException, "a GraphQL API creation is already in progress") {
				r.Retryable = aws.Bool(true)
			}
		}
	})
```

I mention this here because we may be able to remove the wrapping `retryOnAwsCode` calls below in the `aws_graphql_resolver` resource by adding other API calls with their appropriate error messaging to the above logic. If you are interested in simplifying the code after this pull request, we'd happily accept that. 👍